ci/set_github_status.sh

```diff
@@ -8,7 +9,8 @@ function submit() {
         --url "https://api.github.com/repos/gridtools/gt4py/statuses/${commit_sha}" \
         --header 'Content-Type: application/json' \
         --header "authorization: Bearer ${GITHUB_TOKEN}" \
-        --data "{ \"state\": \"${commit_status}\", \"target_url\": \"${CI_PIPELINE_URL}\", \"description\": \"All Gitlab pipelines\", \"context\": \"ci/gitlab/full-pipeline\" }"
+        --data "{ \"state\": \"${commit_status}\", \"target_url\": \"${CI_PIPELINE_URL}\", \"description\": \"All Gitlab pipelines\", \"context\": \"ci/gitlab/full-pipeline\" }" | \
+        jq 'error(.message) // .'
```

Comment:
If you extract potential errors here I would also handle them here.

Reply:
That's what this does, it exits nonzero, which should be propagated when the script exits with `$?`.

Reply:
That's destined to break. If you look at the code from the above comment you see why. If you don't want to handle the error here you should at least return it such that it is apparent to belong to the expected behavior of the function.

Reply:
See above:
> The new version uses set -e to error out as soon as any command returns nonzero, from the log it will be apparent which command that was.
`jq` prints out the error message, the script exits. What more error handling is needed?